### PR TITLE
update the rules for diagnostic setting

### PR DESCRIPTION
**Pull request by shahinshirvani**

### CODE REVIEW

File: azure/iac/diagnosticsettings.rego [hunk 1]
```diff
@@ -8,15 +8,34 @@ package rule
...
+    not contains(lower(resource.scope), "microsoft.keyvault/vaults")
+    metadata:= {
+        "resource_path": [["resources",i,"scope"]]
+    }
+}
+
 azure_attribute_absence["log_keyvault"] {
     resource := input.resources[_]
-    lower(resource.type) == "microsoft.keyvault/vaults/providers/diagnosticsettings"
+    lower(resource.type) == "microsoft.insights/diagnosticsettings"
```
Comment (rezoan) [inline azure/iac/diagnosticsettings.rego:32]: @shahinshirvani can you please check when we add `diagnosticsettings `as a child resource to `microsoft.keyvault/vaults`, how we are writing the resource type of `diagnosticsettings `? 
do we need to write as `microsoft.insights/diagnosticsettings` or `diagnosticsettings` or `microsoft.keyvault/vaults/providers/diagnosticsettings`

I cant remember but did saw `microsoft.keyvault/vaults/providers/diagnosticsettings` somewhere.

**Review by rezoan — CHANGES_REQUESTED** [review]

File: azure/iac/diagnosticsettings.rego [hunk 2]
```diff
@@ -25,43 +44,43 @@ source_path[{"log_keyvault":metadata}] {
 
 azure_issue["log_keyvault"] {
     resource := input.resources[_]
-    lower(resource.type) == "microsoft.keyvault/vaults/providers/diagnosticsettings"
+    lower(resource.type) == "microsoft.insights/diagnosticsettings"
     log := resource.properties.logs[_]
-    lower(log.category) == "auditevent"
-    log.enabled == false
+    log.enabled != true
```
[the remark this answers is not in the document]
Comment (rezoan) [reply azure/iac/diagnosticsettings.rego:50]: @shahinshirvani as discussed with @farchide we don't need to check the category and enabled property as log is an array and possibility that one can be enabled and other can be disabled. which will mislead us. . we just need to make sure this `diagnosticsettings` is associated with this specific `keyvault`

[the remark this answers is not in the document]
Comment (rezoan) [reply azure/iac/diagnosticsettings.rego:50]: @shahinshirvani update according to the comment as well..

**Review by rezoan — CHANGES_REQUESTED** [review]

File: azure/iac/diagnosticsettings.rego [hunk 3]
```diff
@@ -8,15 +8,34 @@ package rule
 
 default log_keyvault = null
 
+azure_attribute_absence ["log_keyvault"] {
+    count([c | lower(input.resources[_].type) == "microsoft.keyvault/vaults"; c := 1]) != count([c | lower(input.resources[_].type) == "microsoft.insights/diagnosticsettings"; c := 1])
```
[the remark this answers is not in the document]
Comment (shahinshirvani) [reply azure/iac/diagnosticsettings.rego]: fixed

File: azure/iac/diagnosticsettings.rego [hunk 4]
```diff
@@ -8,15 +8,34 @@ package rule
...
+}
+
+azure_attribute_absence["log_keyvault"] {
```
[the remark this answers is not in the document]
Comment (shahinshirvani) [reply azure/iac/diagnosticsettings.rego:15]: as we talked ```azure_attribute_absence``` is correct

File: azure/iac/diagnosticsettings.rego (hunk 2 above)
[the remark this answers is not in the document]
Comment (shahinshirvani) [reply azure/iac/diagnosticsettings.rego]: fixed

File: azure/iac/diagnosticsettings.rego (hunk 1 above)
Comment (shahinshirvani) [reply azure/iac/diagnosticsettings.rego:32]: I will check later in a separate issue

**Review by rezoan — APPROVED** [review]

**Pull request merged**